lib/util/string_conv/string_conv.c

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <ctype.h>
+#include <string.h>
+#include <stdlib.h>
+#include <errno.h>
+
+static size_t whitespace_trim(char *out, size_t len, const char *str)
+{
+	if (len == 0) {
+		return 0;
+	}
+
+	const char *end;
+	size_t out_size;
+
+	while (!strncmp(str, " ", 1)) {
+		str++;
+	}
+
+	if (*str == 0) {
+		*out = 0;
+		return 1;
+	}
+
+	end = str + strlen(str) - 1;
+	while (end > str && !strncmp(end, " ", 1)) {
+		end--;
+	}
+	end++;
+
+	out_size = (end - str) + 1;
+
+	if (out_size > len) {
+		return 0;
+	}
+
+	memcpy(out, str, out_size - 1);
+	out[out_size - 1] = 0;
+
+	return out_size;
+}
+
+int string_conv_str2long(char *str, long *val)
+{
+	char trimmed_buf[12] = { 0 };
+	size_t len = whitespace_trim(trimmed_buf, sizeof(trimmed_buf), str);
+
+	if (len < 2) {
+		return -EINVAL;
+	}
+
+	long temp_val;
+	int idx = 0;
+
+	if (!strncmp(trimmed_buf, "-", 1) || !strncmp(trimmed_buf, "+", 1)) {
```

Comment:
Calling `strncmp` where `n` is literal 1 is waste of call. This basically builds up entire function call to just do `return !(trimmed_buf[0] == '-')`